version: "3.9"

networks:
  lucy_network:

services:
  mysql:
    container_name: lucy_mysql
    image: mysql:8.1
    restart: on-failure
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_USER=${DATABASE_USER}
      - MYSQL_PASSWORD=${DATABASE_PASS}
      - MYSQL_DATABASE=${DATABASE_DB}
    volumes:
      - ./data/mysql:/var/lib/mysql
    networks:
      - lucy_network
    ports:
      - 15006:3306
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 10s
      retries: 10
